calculator.c

```diff
@@ -0,0 +1,141 @@
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+int numberStack[50];
+int nSTop = -1;
+int operatorStack[50];
+int opSTop = -1;
+
+int evaluateExp(char exp[]);
+void calculate();
+int operatorPrecedence(char operator);
+int custom_isdigit(char ch);
+int custom_isspace(char ch);
+
+int main()
+{
+    char exp[80];
+    int result = 0;
+    fgets(exp, sizeof(exp), stdin);
+    exp[strcspn(exp, "\n")] = '\0';
+    if (strlen(exp) == 0)
+    {
+        printf("Error: Invalid expression.\n");
+        return 0;
+    }
+    result = evaluateExp(exp);
+    printf("%d\n", result);
+    return 0;
+}
+
+int evaluateExp(char exp[])
+{
+    for (int i = 0; i < strlen(exp); i++)
+    {
+        char ch = exp[i];
+        if (custom_isdigit(ch))
+        {
+            int num = 0;
+            while (custom_isdigit(ch))
+            {
+                num = num * 10 + (ch - '0');
+                i++;
+                if (i < strlen(exp))
+                {
+                    ch = exp[i];
+                }
+                else
+                {
+                    break;
+                }
+            }
+            i--;
+            if (nSTop >= 49)
```

Comment:
Why this condition is there and how did you come with 49?

Reply:
The condition nSTop >= 49 ensures that the stack doesn't overflow. Since we have a fixed stack size of 50, the top index cannot exceed 49 (since the index starts from 0)